backend/src/controller/modelController.ts

```diff
@@ -13,14 +17,32 @@ function handleSetModel(req: OpenAiSetModelRequest, res: Response) {
 	const { model } = req.body;
 
 	if (model === undefined) {
-		res.status(400).send();
-	} else {
-		const configuration =
-			req.body.configuration ?? req.session.chatModel.configuration;
-		req.session.chatModel = { id: model, configuration };
-		console.debug('GPT model set:', JSON.stringify(req.session.chatModel));
-		res.status(200).send();
+		sendErrorResponse(res, 400, 'Missing model');
+		return;
+	}
+
+	if (!chatModelIds.includes(model)) {
+		sendErrorResponse(res, 400, 'Invalid model');
+		return;
 	}
+
+	const configuration =
+		req.body.configuration ?? req.session.chatModel.configuration;
+	req.session.chatModel = { id: model, configuration };
+	console.debug('GPT model set:', JSON.stringify(req.session.chatModel));
+
+	const chatInfoMessage = {
+		infoMessage: `changed model to ${model}`,
+		chatMessageType: 'GENERIC_INFO',
+	} as ChatInfoMessage;
+	// for now, the chat model only changes in the sandbox level
+	req.session.levelState[LEVEL_NAMES.SANDBOX].chatHistory =
```

Comment:
Yep that's fair. Would it help to send the level as part of the request, or can we leave that for another day / an enhancement when needed?

Reply:
it would be nicer and more consistent, but would make no real difference. I wasn't gonna bother